source/fab/source_tree.py

```diff
@@ -22,36 +22,56 @@ def visit(self, candidate: Path):
 
 
 class ExtensionVisitor(TreeVisitor):
-    def __init__(self, extension_map: Mapping[str, Analyser]):
+    def __init__(self,
+                 extension_map: Mapping[str, Union[Type[Task], Type[Command]]],
+                 command_flags_map: Mapping[Type[Command], List[str]],
+                 state: SqliteStateDatabase, workspace: Path):
         self._extension_map = extension_map
+        self._command_flags_map = command_flags_map
+        self._state = state
+        self._workspace = workspace
 
-    def visit(self, candidate: Path):
+    def visit(self, candidate: Path) -> List[Path]:
```

Comment:
You've changed the call signature of this implementation but not of the abstract it's overriding.